### DISCUSSION TITLE
Go module for downloading or locating HashiCorp binaries

### DISCUSSION THREAD

**Discussion by yermulnik** [warrensbox/terraform-switcher#412] [Ideas]
Would it make sense to see whether `hc-install` Go module could replace all of our own code locating/downloading TF binary and verifying signatures/checksums?

https://pkg.go.dev/github.com/hashicorp/hc-install#section-readme

> An **experimental** Go module for downloading or locating HashiCorp binaries, verifying signatures and checksums, and asserting version constraints.
>
> This module is a successor to tfinstall, available in pre-1.0 versions of [terraform-exec](https://github.com/hashicorp/terraform-exec). Current users of tfinstall are advised to move to hc-install before upgrading [terraform-exec](https://github.com/hashicorp/terraform-exec) to v1.0.0.

@warrensbox @MatrixCrawler @crablab @MatthewJohn @jukie

**Comment by MatrixCrawler** [comment]
We should have a look into that. But it states that it is experimental at the moment.
But this could definitely be worth a try. We could save some (or a lot of) code.

**Reply by MatthewJohn** [reply]
I imagine the main issue would be (not that it's merged yet), how it plays with OpenTofu (completely incompatible, I'd expect ;) )

**Reply by yermulnik** [reply]
I knew this whole OpenTofu thing was going to be a hiccup 🤣
